dbms/src/Operators/BucketInput.cpp

```diff
@@ -62,4 +62,29 @@ Block BucketInput::moveOutput()
     output.reset();
     return ret;
 }
+
+Int32 BucketInput::getMinBucketNum(const BucketInputs & inputs)
+{
+    assert(!inputs.empty());
+    Int32 min_bucket_num = NUM_BUCKETS;
+    for (const auto & input : inputs)
+    {
+        if (input.hasOutput())
+            min_bucket_num = std::min(input.bucketNum(), min_bucket_num);
```

Comment:
Why we need to get minimal bucket num and what does bucket number is used for?

Reply:
```
 * ┌──────────────────────────────────────────────────┐
 * │  {bucket0, bucket1, ... bucket255}spilled_file0──┼─►LoadBucketTask1───┐
 * │  {bucket0, bucket1, ... bucket255}spilled_file1──┼─►LoadBucketTask2───┤
 * │  {bucket0, bucket1, ... bucket255}spilled_file2──┼─►LoadBucketTask3───┤
 * │  ...                                             │  ...               │
 * │  {bucket0, bucket1, ... bucket255}spilled_filen──┼─►LoadBucketTaskn───┤
 * └──────────────────────────────────────────────────┘                    │
```
agg restore merges bucket data with the same bucket num from multiple files, and the buckets inside the files are sorted according to their bucket num. So, here we obtain the current smallest bucket num each time, and merge the bucket data of that bucket num.

Reply:
https://github.com/pingcap/tiflash/pull/7318/files#diff-b54d5b998f3a9de97d9452c7c40a8016e79ecbc2251784e96ac0fd6ee840fa16R73

Reply:
> ```
>  * ┌──────────────────────────────────────────────────┐
>  * │  {bucket0, bucket1, ... bucket255}spilled_file0──┼─►LoadBucketTask1───┐
>  * │  {bucket0, bucket1, ... bucket255}spilled_file1──┼─►LoadBucketTask2───┤
>  * │  {bucket0, bucket1, ... bucket255}spilled_file2──┼─►LoadBucketTask3───┤
>  * │  ...                                             │  ...               │
>  * │  {bucket0, bucket1, ... bucket255}spilled_filen──┼─►LoadBucketTaskn───┤
>  * └──────────────────────────────────────────────────┘                    │
> ```
> 
> agg restore merges bucket data with the same bucket num from multiple files, and the buckets inside the files are sorted according to their bucket num. So, here we obtain the current smallest bucket num each time, and merge the bucket data of that bucket num.

get it